src/pages/VoucherRedemption/Confirm.tsx

```diff
@@ -62,54 +65,80 @@ const Amount = (props: Props) => {
       dispatch({ type: SET_AMOUNT, payload: gift_card_detail.amount });
       setLoading(false);
       setView(3);
-    } catch (e) {
-      console.log('error: ', e);
+    } catch (err) {
+      console.log('error: ', err);
       setLoading(false);
     }
   };
 
   return (
     <Container>
       <AmountContainer>
-        <Text width="100%" align="flex-start" onClick={(e) => setView(1)}>
-          {`< Back`}
+        <Text
+          width="100%"
+          align="flex-start"
+          color="#474747"
+          bold="true"
+          wide
+          onClick={(e) => setView(voucher.single_use ? 0 : 1)}
+        >
+          {`< BACK`}
         </Text>
       </AmountContainer>
       <Header>
-        Complete Your Purchase
-        <MoreInfo marginLeft="35px" showShadow={true} />
+        {voucher.single_use ? (
+          <Text size="18px" width="70%">
+            Are you ready to redeem your voucher?
+          </Text>
+        ) : (
+          <Text size="22px">Complete Your Purchase</Text>
+        )}
+        <MoreInfo marginLeft="35px" showShadow={true} inverted={true} />
       </Header>
-      <MessageConatiner>
+      {!voucher.single_use ? (
+        <MessageContainer>
+          <Text size="16px" width="50%">
+            Voucher balance
+          </Text>
+          <Text size="16px" width="50%" align="flex-end">
+            ${(voucher.amount / 100).toFixed(2)}
```

Comment:
maybe extract `(voucher.amount / 100).toFixed(2)` into a `const` since its used in multiple places

Reply:
I realized it's actually only in two places, and it's a slightly different calculation (one's the amount, one's the remainder). But I think there could be room for a better conversion system from simple decimals to the way we store it in our database as an int * 100